piecrust-uplink/src/types.rs

```diff
@@ -40,26 +43,34 @@ pub const MODULE_ID_BYTES: usize = 32;
 pub struct ModuleId([u8; MODULE_ID_BYTES]);
 
 impl ModuleId {
+    /// Creates a placeholder [`ModuleId`] until the host deploys the module
+    /// and sets a real [`ModuleId`]
     pub const fn uninitialized() -> Self {
         ModuleId([0u8; MODULE_ID_BYTES])
     }
 
+    /// Creates a new [`ModuleId`] from an array of bytes
     pub const fn from_bytes(bytes: [u8; MODULE_ID_BYTES]) -> Self {
         Self(bytes)
     }
 
+    /// Returns the array of bytes that make up the [`ModuleId`]
     pub const fn to_bytes(self) -> [u8; MODULE_ID_BYTES] {
         self.0
     }
 
+    /// Returns a reference to the array of bytes that make up the [`ModuleId`]
     pub fn as_bytes(&self) -> &[u8] {
         &self.0
     }
 
+    /// Returns a mutable reference to the array of bytes that make up the
+    /// [`ModuleId`]
     pub fn as_bytes_mut(&mut self) -> &mut [u8] {
         &mut self.0
     }
 
+    /// Determines whether the [`ModuleId`] still is uninitialized
```

Comment:
Perhaps refer to this being usable to determine if we're the first module to be called?

Reply:
See above